pkg/gatherers/workloads/workloads_info.go

```diff
@@ -75,25 +75,54 @@ func gatherWorkloadInfo(
 
 	// load pods in order
 	start := time.Now()
+	limitReached, info, imageCh, err := workloadInfo(ctx, coreClient, imageCh)
+	if err != nil {
+		return []record.Record{}, []error{err}
+	}
+
+	workloadImageResize(info.PodCount)
+
+	records := []record.Record{
+		{
+			Name: "config/workload_info",
+			Item: record.JSONMarshaller{Object: &info},
+		},
+	}
+
+	// wait for as many images as we can find to load
+	handleWorkloadImageInfo(ctx, &info, start, imageCh, imagesDoneCh)
+
+	if limitReached {
+		return records, []error{fmt.Errorf("the %d limit for number of pods gathered was reached", podsLimit)}
+	}
+	return records, nil
+}
+
+// nolint: funlen, gocritic, gocyclo
```

Comment:
I thought the goal was to get rid of the gocyclo - i.e reduce the complexity :)

Reply:
That was the main goal... I reduced it a little bit, the score is better now but... not perfect. Trust me it is better than before and it is almost impossible to remove the remaining complexity without rewrite it completely.